data-prepper-plugins/parquet-codecs/src/test/java/org/opensearch/dataprepper/plugins/codec/parquet/ParquetInputCodecTest.java

```diff
@@ -0,0 +1,202 @@
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.times;
+
+
+
+@ExtendWith(MockitoExtension.class)
+class ParquetInputCodecTest {
+
+    @Mock
+    private Consumer<Record<Event>> eventConsumer;
+
+    private ParquetInputCodec parquetInputCodec;
+    private static FileInputStream fileInputStream;
+
+    private static final String INVALID_PARQUET_INPUT_STREAM = "Invalid Parquet Input Stream";
+
+    @TempDir
+    private static java.nio.file.Path path;
+
+    private ParquetInputCodec createObjectUnderTest() {
+        return new ParquetInputCodec();
+    }
+
+    @BeforeEach
+    void setup() {
+        parquetInputCodec = createObjectUnderTest();
+    }
+
+    @Test
+    void test_when_nullInputStream_then_throwsException() {
+        assertThrows(NullPointerException.class, () ->
+                parquetInputCodec.parse(null, eventConsumer));
+
+        verifyNoInteractions(eventConsumer);
+    }
+
+    @Test
+    void parse_with_null_Consumer_throws()  {
+        parquetInputCodec = createObjectUnderTest();
+
+        final InputStream inputStream = mock(InputStream.class);
+        assertThrows(NullPointerException.class, () ->
+                parquetInputCodec.parse(inputStream, null));
+        verifyNoInteractions(inputStream);
+    }
+
+    @Test
+    void parse_with_empty_InputStream_does_not_call_Consumer() throws IOException {
+        final ByteArrayInputStream emptyInputStream = new ByteArrayInputStream(new byte[]{});
+        createObjectUnderTest().parse(emptyInputStream, eventConsumer);
+        verifyNoInteractions(eventConsumer);
+    }
+
+    @Test
+    public void parse_with_Invalid_InputStream_then_catches_exception() {
+        Consumer<Record<Event>> eventConsumer = mock(Consumer.class);
+        Assertions.assertDoesNotThrow(()->
+                parquetInputCodec.parse(createInvalidParquetStream(),eventConsumer));
+
+    }
+
+    @ParameterizedTest
+    @ValueSource(ints = {1,10,100,1000})
+    void test_when_HappyCaseParquetInputStream_then_callsConsumerWithParsedEvents(final int numberOfRecords) throws IOException {
+        InputStream inputStream = createRandomParquetStream(numberOfRecords);
+
+        parquetInputCodec.parse(inputStream,eventConsumer);
+
+        final ArgumentCaptor<Record<Event>> recordArgumentCaptor = ArgumentCaptor.forClass(Record.class);
+        verify(eventConsumer, times(numberOfRecords)).accept(recordArgumentCaptor.capture());
+        final List<Record<Event>> actualRecords = recordArgumentCaptor.getAllValues();
+        assertThat(actualRecords.size(), equalTo(numberOfRecords));
+
+        for (int i = 0; i < actualRecords.size(); i++) {
+
+            final Record<Event> actualRecord = actualRecords.get(i);
+            assertThat(actualRecord, notNullValue());
+            assertThat(actualRecord.getData(), notNullValue());
+            assertThat(actualRecord.getData().getMetadata(), notNullValue());
```

Comment:
Please assert that the actual records (`actualRecord.getData()` and `actualRecord.getData().getMetadata()` match the inputstream

Reply:
How can we fetch data and metadata from input stream without parsing it into records? Please correct me if I am wrong.